Compile Lapack

Here is how to get started:
```console
git clone https://github.com/Reference-LAPACK/lapack
cd lapack
mkdir build
cd build
FC=gfortran cmake -DCBLAS=no -DLAPACKE=no -DBUILD_TESTING=no -DBUILD_DOUBLE=no -DBUILD_COMPLEX=no -DBUILD_COMPLEX16=no ..
make
```

@meow464 can you post exact instructions? Here is what I tried on macOS:
```console
$ git clone https://github.com/Reference-LAPACK/lapack
$ cd lapack
$ mkdir build
$ cd build
$ cmake .. -DCMAKE_Fortran_COMPILER=lfortran -DCBLAS=no -DLAPACKE=no -DBUILD_TESTING=no -DBUILD_DOUBLE=no -DBUILD_COMPLEX=no -DBUILD_COMPLEX16=no -DLAPACKE_WITH_TMG=no -DCMAKE_Fortran_PREPROCESS=yes -DCMAKE_Fortran_FLAGS="--fixed-form-infer --implicit-interface"

-- The C compiler identification is AppleClang 13.0.0.13000029
-- The CXX compiler identification is AppleClang 13.0.0.13000029
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Library/Developer/CommandLineTools/usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Library/Developer/CommandLineTools/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Setting build type to 'Release' as none was specified.
-- Checking if build type is 'Coverage'
-- Checking if build type is 'Coverage': 0
-- Build tests: no
-- Build deprecated routines: OFF
-- Build single precision real: ON
-- Build double precision real: no
-- Build single precision complex: no
-- Build double precision complex: no
-- Using supplied NETLIB BLAS implementation
-- The Fortran compiler identification is unknown
-- Detecting Fortran compiler ABI info
-- Detecting Fortran compiler ABI info - done
-- Check for working Fortran compiler: /Users/ondrej/bin/lfortran - skipped
CMake Warning at CMAKE/CheckLAPACKCompilerFlags.cmake:236 (message):
  Fortran local arrays should be allocated on the stack.  Please use a
  compiler which guarantees that feature.  See
  https://github.com/Reference-LAPACK/lapack/pull/188 and references therein.
Call Stack (most recent call first):
  BLAS/CMakeLists.txt:5 (CheckLAPACKCompilerFlags)


-- Using supplied NETLIB LAPACK implementation
CMake Warning at CMAKE/CheckLAPACKCompilerFlags.cmake:236 (message):
  Fortran local arrays should be allocated on the stack.  Please use a
  compiler which guarantees that feature.  See
  https://github.com/Reference-LAPACK/lapack/pull/188 and references therein.
Call Stack (most recent call first):
  CMakeLists.txt:322 (CheckLAPACKCompilerFlags)


-- Looking for Fortran NONE - not found
-- Looking for Fortran INT_CPU_TIME - not found
-- Looking for Fortran EXT_ETIME - not found
-- Looking for Fortran EXT_ETIME_ - not found
-- Looking for Fortran INT_ETIME - not found
-- --> Will use second_NONE.f and dsecnd_NONE.f as timing function.
-- Detecting Fortran/C Interface
Failed to compile
-- Verifying Fortran/C Compiler Compatibility
Failed to compile
CMake Warning (dev) at /Users/ondrej/repos/spack/opt/spack/darwin-monterey-m1/apple-clang-13.0.0/cmake-3.23.1-3geuicgdlylivqqft4pjwrdnmb63dfxr/share/cmake-3.23/Modules/FortranCInterface.cmake:309 (message):
  No FortranCInterface mangling known for VerifyFortran
Call Stack (most recent call first):
  /Users/ondrej/repos/spack/opt/spack/darwin-monterey-m1/apple-clang-13.0.0/cmake-3.23.1-3geuicgdlylivqqft4pjwrdnmb63dfxr/share/cmake-3.23/Modules/FortranCInterface/Verify/CMakeLists.txt:16 (FortranCInterface_HEADER)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Verifying Fortran/C Compiler Compatibility - Failed
CMake Error at /Users/ondrej/repos/spack/opt/spack/darwin-monterey-m1/apple-clang-13.0.0/cmake-3.23.1-3geuicgdlylivqqft4pjwrdnmb63dfxr/share/cmake-3.23/Modules/FortranCInterface.cmake:400 (message):
  The Fortran compiler:

    /Users/ondrej/bin/lfortran

  and the C compiler:

    /Library/Developer/CommandLineTools/usr/bin/cc

  failed to compile a simple test project using both languages.  The output
  was:

    Change Dir: /Users/ondrej/repos/lapack/build/CMakeFiles/FortranCInterface/VerifyC
    
    Run Build Command(s):/usr/bin/make -f Makefile VerifyFortranC && /Users/ondrej/repos/spack/opt/spack/darwin-monterey-m1/apple-clang-13.0.0/cmake-3.23.1-3geuicgdlylivqqft4pjwrdnmb63dfxr/bin/cmake -S/Users/ondrej/repos/spack/opt/spack/darwin-monterey-m1/apple-clang-13.0.0/cmake-3.23.1-3geuicgdlylivqqft4pjwrdnmb63dfxr/share/cmake-3.23/Modules/FortranCInterface/Verify -B/Users/ondrej/repos/lapack/build/CMakeFiles/FortranCInterface/VerifyC --check-build-system CMakeFiles/Makefile.cmake 0
    /Library/Developer/CommandLineTools/usr/bin/make  -f CMakeFiles/Makefile2 VerifyFortranC
    /Users/ondrej/repos/spack/opt/spack/darwin-monterey-m1/apple-clang-13.0.0/cmake-3.23.1-3geuicgdlylivqqft4pjwrdnmb63dfxr/bin/cmake -S/Users/ondrej/repos/spack/opt/spack/darwin-monterey-m1/apple-clang-13.0.0/cmake-3.23.1-3geuicgdlylivqqft4pjwrdnmb63dfxr/share/cmake-3.23/Modules/FortranCInterface/Verify -B/Users/ondrej/repos/lapack/build/CMakeFiles/FortranCInterface/VerifyC --check-build-system CMakeFiles/Makefile.cmake 0
    /Users/ondrej/repos/spack/opt/spack/darwin-monterey-m1/apple-clang-13.0.0/cmake-3.23.1-3geuicgdlylivqqft4pjwrdnmb63dfxr/bin/cmake -E cmake_progress_start /Users/ondrej/repos/lapack/build/CMakeFiles/FortranCInterface/VerifyC/CMakeFiles 5
    /Library/Developer/CommandLineTools/usr/bin/make  -f CMakeFiles/Makefile2 CMakeFiles/VerifyFortranC.dir/all
    /Library/Developer/CommandLineTools/usr/bin/make  -f CMakeFiles/VerifyFortran.dir/build.make CMakeFiles/VerifyFortran.dir/depend
    cd /Users/ondrej/repos/lapack/build/CMakeFiles/FortranCInterface/VerifyC && /Users/ondrej/repos/spack/opt/spack/darwin-monterey-m1/apple-clang-13.0.0/cmake-3.23.1-3geuicgdlylivqqft4pjwrdnmb63dfxr/bin/cmake -E cmake_depends "Unix Makefiles" /Users/ondrej/repos/spack/opt/spack/darwin-monterey-m1/apple-clang-13.0.0/cmake-3.23.1-3geuicgdlylivqqft4pjwrdnmb63dfxr/share/cmake-3.23/Modules/FortranCInterface/Verify /Users/ondrej/repos/spack/opt/spack/darwin-monterey-m1/apple-clang-13.0.0/cmake-3.23.1-3geuicgdlylivqqft4pjwrdnmb63dfxr/share/cmake-3.23/Modules/FortranCInterface/Verify /Users/ondrej/repos/lapack/build/CMakeFiles/FortranCInterface/VerifyC /Users/ondrej/repos/lapack/build/CMakeFiles/FortranCInterface/VerifyC /Users/ondrej/repos/lapack/build/CMakeFiles/FortranCInterface/VerifyC/CMakeFiles/VerifyFortran.dir/DependInfo.cmake
    Scanning dependencies of target VerifyFortran
    /Library/Developer/CommandLineTools/usr/bin/make  -f CMakeFiles/VerifyFortran.dir/build.make CMakeFiles/VerifyFortran.dir/build
    [ 20%] Building Fortran object CMakeFiles/VerifyFortran.dir/VerifyFortran.f.o
    /Users/ondrej/bin/lfortran  -I/Users/ondrej/repos/lapack/build/CMakeFiles/FortranCInterface/VerifyC --fixed-form-infer --implicit-interface -c /Users/ondrej/repos/spack/opt/spack/darwin-monterey-m1/apple-clang-13.0.0/cmake-3.23.1-3geuicgdlylivqqft4pjwrdnmb63dfxr/share/cmake-3.23/Modules/FortranCInterface/Verify/VerifyFortran.f -o CMakeFiles/VerifyFortran.dir/VerifyFortran.f.o
    [ 40%] Linking Fortran static library libVerifyFortran.a
    /Users/ondrej/repos/spack/opt/spack/darwin-monterey-m1/apple-clang-13.0.0/cmake-3.23.1-3geuicgdlylivqqft4pjwrdnmb63dfxr/bin/cmake -P CMakeFiles/VerifyFortran.dir/cmake_clean_target.cmake
    /Users/ondrej/repos/spack/opt/spack/darwin-monterey-m1/apple-clang-13.0.0/cmake-3.23.1-3geuicgdlylivqqft4pjwrdnmb63dfxr/bin/cmake -E cmake_link_script CMakeFiles/VerifyFortran.dir/link.txt --verbose=1
    /Library/Developer/CommandLineTools/usr/bin/ar qc libVerifyFortran.a CMakeFiles/VerifyFortran.dir/VerifyFortran.f.o
    /Library/Developer/CommandLineTools/usr/bin/ranlib libVerifyFortran.a
    warning: /Library/Developer/CommandLineTools/usr/bin/ranlib: archive library: libVerifyFortran.a the table of contents is empty (no object file members in the library define global symbols)
    [ 40%] Built target VerifyFortran
    /Library/Developer/CommandLineTools/usr/bin/make  -f CMakeFiles/VerifyFortranC.dir/build.make CMakeFiles/VerifyFortranC.dir/depend
    cd /Users/ondrej/repos/lapack/build/CMakeFiles/FortranCInterface/VerifyC && /Users/ondrej/repos/spack/opt/spack/darwin-monterey-m1/apple-clang-13.0.0/cmake-3.23.1-3geuicgdlylivqqft4pjwrdnmb63dfxr/bin/cmake -E cmake_depends "Unix Makefiles" /Users/ondrej/repos/spack/opt/spack/darwin-monterey-m1/apple-clang-13.0.0/cmake-3.23.1-3geuicgdlylivqqft4pjwrdnmb63dfxr/share/cmake-3.23/Modules/FortranCInterface/Verify /Users/ondrej/repos/spack/opt/spack/darwin-monterey-m1/apple-clang-13.0.0/cmake-3.23.1-3geuicgdlylivqqft4pjwrdnmb63dfxr/share/cmake-3.23/Modules/FortranCInterface/Verify /Users/ondrej/repos/lapack/build/CMakeFiles/FortranCInterface/VerifyC /Users/ondrej/repos/lapack/build/CMakeFiles/FortranCInterface/VerifyC /Users/ondrej/repos/lapack/build/CMakeFiles/FortranCInterface/VerifyC/CMakeFiles/VerifyFortranC.dir/DependInfo.cmake
    /Library/Developer/CommandLineTools/usr/bin/make  -f CMakeFiles/VerifyFortranC.dir/build.make CMakeFiles/VerifyFortranC.dir/build
    [ 60%] Building C object CMakeFiles/VerifyFortranC.dir/main.c.o
    /Library/Developer/CommandLineTools/usr/bin/cc  -I/Users/ondrej/repos/lapack/build/CMakeFiles/FortranCInterface/VerifyC -O3 -DNDEBUG -arch arm64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk -MD -MT CMakeFiles/VerifyFortranC.dir/main.c.o -MF CMakeFiles/VerifyFortranC.dir/main.c.o.d -o CMakeFiles/VerifyFortranC.dir/main.c.o -c /Users/ondrej/repos/spack/opt/spack/darwin-monterey-m1/apple-clang-13.0.0/cmake-3.23.1-3geuicgdlylivqqft4pjwrdnmb63dfxr/share/cmake-3.23/Modules/FortranCInterface/Verify/main.c
    [ 80%] Building C object CMakeFiles/VerifyFortranC.dir/VerifyC.c.o
    /Library/Developer/CommandLineTools/usr/bin/cc  -I/Users/ondrej/repos/lapack/build/CMakeFiles/FortranCInterface/VerifyC -O3 -DNDEBUG -arch arm64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk -MD -MT CMakeFiles/VerifyFortranC.dir/VerifyC.c.o -MF CMakeFiles/VerifyFortranC.dir/VerifyC.c.o.d -o CMakeFiles/VerifyFortranC.dir/VerifyC.c.o -c /Users/ondrej/repos/spack/opt/spack/darwin-monterey-m1/apple-clang-13.0.0/cmake-3.23.1-3geuicgdlylivqqft4pjwrdnmb63dfxr/share/cmake-3.23/Modules/FortranCInterface/Verify/VerifyC.c
    [100%] Linking C executable VerifyFortranC
    /Users/ondrej/repos/spack/opt/spack/darwin-monterey-m1/apple-clang-13.0.0/cmake-3.23.1-3geuicgdlylivqqft4pjwrdnmb63dfxr/bin/cmake -E cmake_link_script CMakeFiles/VerifyFortranC.dir/link.txt --verbose=1
    /Library/Developer/CommandLineTools/usr/bin/cc -O3 -DNDEBUG -arch arm64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk -Wl,-search_paths_first -Wl,-headerpad_max_install_names CMakeFiles/VerifyFortranC.dir/main.c.o CMakeFiles/VerifyFortranC.dir/VerifyC.c.o -o VerifyFortranC  libVerifyFortran.a 
    Undefined symbols for architecture arm64:
      "_VerifyFortran", referenced from:
          _main in main.c.o
    ld: symbol(s) not found for architecture arm64
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
    make[3]: *** [VerifyFortranC] Error 1
    make[2]: *** [CMakeFiles/VerifyFortranC.dir/all] Error 2
    make[1]: *** [CMakeFiles/VerifyFortranC.dir/rule] Error 2
    make: *** [VerifyFortranC] Error 2
    
    
Call Stack (most recent call first):
  LAPACKE/include/CMakeLists.txt:17 (FortranCInterface_VERIFY)


-- Configuring incomplete, errors occurred!
See also "/Users/ondrej/repos/lapack/build/CMakeFiles/CMakeOutput.log".
See also "/Users/ondrej/repos/lapack/build/CMakeFiles/CMakeError.log".
```

Thansk @certik for the feedback. The lower/upper was indeed a false track. It happens that cmake fortran detection code tries hard to find which mangling is used by the underlying fortran compiler, so none of `--all-mangling --mangle-underscore --bindc-mangling` is actually needed. But cmake test code uses the C preprocessor, so `--cpp-infer` is required. Then ` --implicit-interface` has to be used because some cmake test code references implictly defined functions.

So far I'm stuck with `undefined reference to `_lfortran_printf'` which I can workaround with ` -llfortran_runtime_static -lm` but I still need to find a good non intrusive way to pass this to cmake.

I'm still getting the linking error regarding VerifyFortranC with gcc, but clang works.
```
gcc -DNDEBUG -L"/home/tuco/Projects/LFortran/lfortran/inst_version_macros/bin/../lib" -Wl,-rpath,"/home/tuco/Projects/LFortran/lfortran/inst_version_macros/bin/../lib" -llfortran_runtime -lm CMakeFiles/VerifyFortranC.dir/main.c.o CMakeFiles/VerifyFortranC.dir/VerifyC.c.o -o VerifyFortranC  libVerifyFortran.a

/usr/bin/ld: libVerifyFortran.a(VerifyFortran.f.o): in function `verifyfortran':
LFortran:(.text+0x19): undefined reference to `_lfortran_printf'
collect2: error: ld returned 1 exit status
```

> Great job. I think you built an empty library --- all the object files are empty. So you tested compiling to ASR, which succeeded, and that's great news. Now we need to test the LLVM backend. Use `--generate-object-file` option and see.

I tried using `--generate-object-code` as a part of `-DCMAKE_Fortran_FLAGS` as:
```bash
$ cmake .. -DCMAKE_Fortran_COMPILER=lfortran -DCBLAS=no -DLAPACKE=no -DBUILD_TESTING=no -DBUILD_DOUBLE=no -DBUILD_COMPLEX=no -DBUILD_COMPLEX16=no -DLAPACKE_WITH_TMG=no -DCMAKE_Fortran_PREPROCESS=yes -DCMAKE_Fortran_FLAGS="--fixed-form-infer --implicit-interface --generate-object-code" -DCMAKE_INSTALL_LIBDIR=$HOME/ext/lapack
```

I got the below codegen error on executing (on macOS M1):
```bash
[ 33%] Building Fortran object SRC/CMakeFiles/lapack.dir/sgedmdq.f90.o
[ 33%] Building Fortran object SRC/CMakeFiles/lapack.dir/slaed0.f.o
[ 34%] Building Fortran object SRC/CMakeFiles/lapack.dir/sgetrf.f.o
[ 35%] Building Fortran object SRC/CMakeFiles/lapack.dir/sbdsqr.f.o
[ 35%] Building Fortran object SRC/CMakeFiles/lapack.dir/slaed3.f.o
[ 35%] Building Fortran object SRC/CMakeFiles/lapack.dir/slarre.f.o
[ 35%] Building Fortran object SRC/CMakeFiles/lapack.dir/slarrf.f.o
[ 35%] Building Fortran object SRC/CMakeFiles/lapack.dir/slasd0.f.o
[ 35%] Building Fortran object SRC/CMakeFiles/lapack.dir/slasd6.f.o
[ 35%] Building Fortran object SRC/CMakeFiles/lapack.dir/slaed8.f.o
[ 35%] Building Fortran object SRC/CMakeFiles/lapack.dir/slaed2.f.o
[ 36%] Building Fortran object SRC/CMakeFiles/lapack.dir/slasda.f.o
[ 36%] Building Fortran object SRC/CMakeFiles/lapack.dir/sstedc.f.o
[ 36%] Building Fortran object SRC/CMakeFiles/lapack.dir/sstevx.f.o
[ 36%] Building Fortran object SRC/CMakeFiles/lapack.dir/slarrd.f.o
...
...
return:                                           ; preds = %ifcont
  %5 = load i32, i32* %_lcompilers_min0_i323, align 4
  ret i32 %5
}

define i32 @_lcompilers_min0_i324(i32* %x0, i32* %x1) {
.entry:
  %_lcompilers_min0_i324 = alloca i32, align 4
  %0 = load i32, i32* %x0, align 4
  store i32 %0, i32* %_lcompilers_min0_i324, align 4
  %1 = load i32, i32* %x1, align 4
  %2 = load i32, i32* %_lcompilers_min0_i324, align 4
  %3 = icmp slt i32 %1, %2
  br i1 %3, label %then, label %else

then:                                             ; preds = %.entry
  %4 = load i32, i32* %x1, align 4
  store i32 %4, i32* %_lcompilers_min0_i324, align 4
  br label %ifcont

else:                                             ; preds = %.entry
  br label %ifcont

ifcont:                                           ; preds = %else, %then
  br label %return

return:                                           ; preds = %ifcont
  %5 = load i32, i32* %_lcompilers_min0_i324, align 4
  ret i32 %5
}

define i32 @_lcompilers_min0_i325(i32* %x0, i32* %x1) {
.entry:
  %_lcompilers_min0_i325 = alloca i32, align 4
  %0 = load i32, i32*code generation error: Type mismatch in subroutine call, expected `f32[:,:]`, passed `f32`
   --> /Users/gxyd/OpenSource/lfortran/lapack/SRC/slaorhr_col_getrfnp2.f:294:47
    |
294 |          CALL SLAORHR_COL_GETRFNP2( M-N1, N2, A( N1+1, N1+1 ), LDA,
    |                                               ^^^^^^^^^^^^^^^ 


Note: Please report unclear, confusing or incorrect messages as bugs at
https://github.com/lfortran/lfortran/issues.
make[2]: *** [SRC/CMakeFiles/lapack.dir/build.make:4690: SRC/CMakeFiles/lapack.dir/slaorhr_col_getrfnp2.f.o] Error 5
code generation error: Type mismatch in subroutine call, expected `f32[:,:]`, passed `f32`
   --> /Users/gxyd/OpenSource/lfortran/lapack/SRC/strti2.f:199:28
    |
199 |      $                     A( J+1, J+1 ), LDA, A( J+1, J ), 1 )
    |                            ^^^^^^^^^^^^^ 


Note: Please report unclear, confusing or incorrect messages as bugs at
https://github.com/lfortran/lfortran/issues.
 %x0, align 4
  store i32 %0, i32* %_lcompilers_min0_i325, align 4
  %1 = load i32, i32* %x1, align 4
  %2 = load i32, i32* %_lcompilers_min0_i325, align 4
  %3 = icmp slt i32 %1, %2
  br i1 %3, label %then, label %else

then:                                             ; preds = %.entry
  %4 = load i32, i32* %x1, align 4
  store i32 %4, i32* %_lcompilers_min0_i325, align 4
  br label %ifcont

else:                                             ; preds = %.entry
  br label %ifcont

ifcont:                                           ; preds = %else, %then
  br label %return

return:                                           ; preds = %ifcont
  %5 = load i32, i32* %_lcompilers_min0_i325, align 4
  ret i32 %5
}

define i32 @_lcompilers_min0_i326(i32* %x0, i32* %x1) {
.entry:
  %_lcompilers_min0_i326 = alloca i32, align 4
; ModuleID = 'LFortran'
source_filename = "LFortran"

%array = type { float*, i32, %dimension_descriptor*, i1, i32 }
%dimension_descriptor = type { i32, i32, i32 }

@0 = private unnamed_addr constant [7 x i8] c"SGERQF\00", align 1
@1 = private unnamed_addr constant [2 x i8] c" \00", align 1
@2 = private unnamed_addr constant [7 x i8] c"STZRZF\00", align 1
@3 = private unnamed_addr constant [7 x i8] c"SGERQF\00", align 1
@4 = private unnamed_addr constant [2 x i8] c" \00", align 1
@5 = private unnamed_addr constant [7 x i8] c"SGERQF\00", align 1
@6 = private unnamed_addr constant [2 x i8] c" \00", align 1
@7 = private unnamed_addr constant [9 x i8] c"B  %0 = load i32, i32* %x0, align 4
  store i32 %0, i32* %_lcompilers_min0_i326, align 4
  %1 = load i32, i32* %x1, align 4
  %2 = load i32, i32* %_lcompilers_min0_i326, align 4
  %3 = icmp slt i32 %1, %2
  br i1 %3, label %then, label %else

then:                                             ; preds = %.entry
  %4 = load i32, i32* %x1, align 4
  store i32 %4, i32* %_lcompilers_min0_i326, align 4
  br label %ifcont

else:                                             ; preds = %.entry
  br label %ifcont

ifcont:                                           ; preds = %else, %then
  br label %return

return:                                           ; preds = %ifcont
  %5 = load i32, i32* %_lcompilers_min0_i326, align 4
  ret i32 %5
}

define i32 @_lcompilers_min0_i327(i32* %x0, i32* %x1) {
.entry:
  %_lcompilers_min0_i327 = alloca i32, align 4
  %0 = load i32, i32* %x0, align 4
  store i32 %0, i32* %_lcompilers_min0_i327, align 4
  %1 = load i32, i32* %x1, align 4
  %2 = load i32, i32* %_code generation error: Type mismatch in subroutine call, expected `f32[:]`, passed `f32`
   --> /Users/gxyd/OpenSource/lfortran/lapack/SRC/stptri.f:226:28
    |
226 |      $                     AP( JCLAST ), AP( JC+1 ), 1 )
    |                            ^^^^^^^^^^^^ 
```


I've opened an issue for it here: https://github.com/lfortran/lfortran/issues/4852

Following the discussion in https://github.com/lfortran/lfortran/issues/4852 the above error that I got was because of missing `--legacy-array-sections` when building LAPACK.

Now, below 


```console
$ cmake .. -DCMAKE_Fortran_COMPILER=lfortran -DCBLAS=no -DLAPACKE=no -DBUILD_TESTING=no -DBUILD_DOUBLE=no -DBUILD_COMPLEX=no -DBUILD_COMPLEX16=no -DLAPACKE_WITH_TMG=no -DCMAKE_Fortran_PREPROCESS=yes -DCMAKE_Fortran_FLAGS="--fixed-form-infer --implicit-interface --generate-object-code --legacy-array-sections" -DCMAKE_INSTALL_LIBDIR=$HOME/ext/lapack
$ cmake --build . -j --target install

...
...
[ 88%] Building Fortran object SRC/CMakeFiles/lapack.dir/slanst.f.o
[ 88%] Building Fortran object SRC/CMakeFiles/lapack.dir/slaed8.f.o
[ 88%] Building Fortran object SRC/CMakeFiles/lapack.dir/slarra.f.o
[ 88%] Building Fortran object SRC/CMakeFiles/lapack.dir/slarrj.f.o
[ 88%] Building Fortran object SRC/CMakeFiles/lapack.dir/slarnv.f.o
[ 88%] Building Fortran object SRC/CMakeFiles/lapack.dir/slascl.f.o
[ 88%] Building Fortran object SRC/CMakeFiles/lapack.dir/slaruv.f.o
[ 89%] Building Fortran object SRC/CMakeFiles/lapack.dir/slarrb.f.o
[ 89%] Building Fortran object SRC/CMakeFiles/lapack.dir/slarrc.f.o
[ 89%] Building Fortran object SRC/CMakeFiles/lapack.dir/slarre.f.o
[ 89%] Building Fortran object SRC/CMakeFiles/lapack.dir/slarrd.f.o
[ 90%] Building Fortran object SRC/CMakeFiles/lapack.dir/slarrk.f.o
[ 90%] Building Fortran object SRC/CMakeFiles/lapack.dir/slasd6.f.o
[ 90%] Building Fortran object SRC/CMakeFiles/lapack.dir/slaed6.f.o
[ 90%] Building Fortran object SRC/CMakeFiles/lapack.dir/slarrf.f.o
[ 90%] Building Fortran object SRC/CMakeFiles/lapack.dir/slasd3.f.o
[ 90%] Building Fortran object SRC/CMakeFiles/lapack.dir/slaneg.f.o
[ 91%] Building Fortran object SRC/CMakeFiles/lapack.dir/slas2.f.o
[ 91%] Building Fortran object SRC/CMakeFiles/lapack.dir/slasd2.f.o
[ 91%] Building Fortran object SRC/CMakeFiles/lapack.dir/slasq1.f.o
[ 91%] Building Fortran object SRC/CMakeFiles/lapack.dir/slarrr.f.o
[ 91%] Building Fortran object SRC/CMakeFiles/lapack.dir/slasd1.f.o
[ 92%] Building Fortran object SRC/CMakeFiles/lapack.dir/slasd4.f.o
[ 93%] Building Fortran object SRC/CMakeFiles/lapack.dir/slasda.f.o
[ 93%] Building Fortran object SRC/CMakeFiles/lapack.dir/slasd8.f.o
[ 93%] Building Fortran object SRC/CMakeFiles/lapack.dir/slasd5.f.o
[ 93%] Building Fortran object SRC/CMakeFiles/lapack.dir/slasdq.f.o
[ 93%] Building Fortran object SRC/CMakeFiles/lapack.dir/slasd0.f.o
[ 93%] Building Fortran object SRC/CMakeFiles/lapack.dir/slaset.f.o
[ 93%] Building Fortran object SRC/CMakeFiles/lapack.dir/slasdt.f.o
[ 93%] Building Fortran object SRC/CMakeFiles/lapack.dir/slasd7.f.o
[ 94%] Building Fortran object SRC/CMakeFiles/lapack.dir/slasv2.f.o
[ 94%] Building Fortran object SRC/CMakeFiles/lapack.dir/slasq2.f.o
[ 94%] Building Fortran object SRC/CMakeFiles/lapack.dir/spttrf.f.o
[ 94%] Building Fortran object SRC/CMakeFiles/lapack.dir/slasq5.f.o
[ 94%] Building Fortran object SRC/CMakeFiles/lapack.dir/slasq4.f.o
[ 94%] Building Fortran object SRC/CMakeFiles/lapack.dir/slasq6.f.o
[ 94%] Building Fortran object SRC/CMakeFiles/lapack.dir/slasq3.f.o
[ 94%] Building Fortran object SRC/CMakeFiles/lapack.dir/sstein.f.o
[ 94%] Building Fortran object SRC/CMakeFiles/lapack.dir/ssterf.f.o
[ 94%] Building Fortran object SRC/CMakeFiles/lapack.dir/sstevx.f.o
[ 94%] Building Fortran object SRC/CMakeFiles/lapack.dir/sstebz.f.o
[ 96%] Building Fortran object SRC/CMakeFiles/lapack.dir/ssteqr.f.o
[ 96%] Building Fortran object SRC/CMakeFiles/lapack.dir/slasrt.f.o
[ 96%] Building Fortran object SRC/CMakeFiles/lapack.dir/sstedc.f.o
[ 96%] Building Fortran object SRC/CMakeFiles/lapack.dir/slasr.f.o
[ 97%] Building Fortran object SRC/CMakeFiles/lapack.dir/iparmq.f.o
[ 97%] Building Fortran object SRC/CMakeFiles/lapack.dir/slartgs.f.o
[ 97%] Building Fortran object SRC/CMakeFiles/lapack.dir/ilaprec.f.o
[ 97%] Building Fortran object SRC/CMakeFiles/lapack.dir/ilaenv2stage.f.o
[ 97%] Building Fortran object SRC/CMakeFiles/lapack.dir/ilaenv.f.o
[ 97%] Building Fortran object SRC/CMakeFiles/lapack.dir/slartgp.f.o
[ 97%] Building Fortran object SRC/CMakeFiles/lapack.dir/lsamen.f.o
[ 97%] Building Fortran object SRC/CMakeFiles/lapack.dir/__/INSTALL/second_NONE.f.o
[ 98%] Building Fortran object SRC/CMakeFiles/lapack.dir/__/INSTALL/sroundup_lwork.f.o
[ 98%] Building Fortran object SRC/CMakeFiles/lapack.dir/xerbla.f.o
[ 98%] Building Fortran object SRC/CMakeFiles/lapack.dir/ieeeck.f.o
[ 98%] Building Fortran object SRC/CMakeFiles/lapack.dir/__/INSTALL/slamch.f.o
[ 98%] Building Fortran object SRC/CMakeFiles/lapack.dir/iladiag.f.o
[ 98%] Building Fortran object SRC/CMakeFiles/lapack.dir/slartg.f90.o
[ 98%] Building Fortran object SRC/CMakeFiles/lapack.dir/ilauplo.f.o
[ 98%] Building Fortran object SRC/CMakeFiles/lapack.dir/ilatrans.f.o
[ 98%] Building Fortran object SRC/CMakeFiles/lapack.dir/xerbla_array.f.o
[ 99%] Building Fortran object SRC/CMakeFiles/lapack.dir/__/INSTALL/ilaver.f.o
[ 99%] Building Fortran object SRC/CMakeFiles/lapack.dir/__/INSTALL/lsame.f.o
style suggestion: Use '>=' instead of '.ge.'
   --> /Users/gxyd/OpenSource/lfortran/lapack/SRC/sgedmd.f90:644:27
    |
644 |                IF ( SCALE .GE. (OFL / ROOTSC) ) THEN
    |                           ^^^^ help: write this as '>='

style suggestion: Use '>=' instead of '.ge.'
   --> /Users/gxyd/OpenSource/lfortran/lapack/SRC/sgedmd.f90:717:27
    |
717 |                IF ( SCALE .GE. (OFL / ROOTSC) ) THEN
    |                           ^^^^ help: write this as '>='


Note: Please report unclear, confusing or incorrect messages as bugs at
https://github.com/lfortran/lfortran/issues.
LCOMPILERS_ASSERT failed: /Users/gxyd/OpenSource/lfortran/src/libasr/codegen/asr_to_llvm.cpp
function visit_ArrayBound(), line number 10062 at 
false
/bin/sh: line 1:  8703 Abort trap: 6           /Users/gxyd/OpenSource/lfortran/src/bin/lfortran --fixed-form-infer --implicit-interface --generate-object-code --legacy-array-sections -c /Users/gxyd/OpenSource/lfortran/lapack/SRC/stfsm.f -o CMakeFiles/lapack.dir/stfsm.f.o
make[2]: *** [SRC/CMakeFiles/lapack.dir/build.make:4040: SRC/CMakeFiles/lapack.dir/stfsm.f.o] Error 134
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:214: SRC/CMakeFiles/lapack.dir/all] Error 2
make: *** [Makefile:156: all] Error 2
```

I've opened an issue with MRE for the same here: https://github.com/lfortran/lfortran/issues/4924

i tried compiling, here is how to do it:

```
git clone https://github.com/Reference-LAPACK/lapack
cd lapack
```

Apply the following diff:

```diff
diff --git a/BLAS/SRC/CMakeLists.txt b/BLAS/SRC/CMakeLists.txt
index b9e6f7c4a..8bf4f9545 100644
--- a/BLAS/SRC/CMakeLists.txt
+++ b/BLAS/SRC/CMakeLists.txt
@@ -126,7 +126,7 @@ if(BUILD_INDEX64_EXT_API)
     if(CMAKE_Fortran_COMPILER_ID STREQUAL "NAG")
       set_source_files_properties(${F} PROPERTIES COMPILE_FLAGS "-fpp")
     else()
-      set_source_files_properties(${F} PROPERTIES COMPILE_FLAGS "-cpp")
+      set_source_files_properties(${F} PROPERTIES COMPILE_FLAGS "--cpp")
     endif()
     file(STRINGS ${F} ${F}.lst)
     list(FILTER ${F}.lst INCLUDE REGEX "subroutine|SUBROUTINE|external|EXTERNAL|function|FUNCTION")
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 59aaed152..d7785c120 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -380,7 +380,7 @@ unset(LAPACK_INSTALL_EXPORT_NAME_CACHE)
 add_subdirectory(LAPACKE/include)
 
 if(LAPACKE)
-  add_subdirectory(LAPACKE)
+  # add_subdirectory(LAPACKE)
 endif()
 
 
diff --git a/LAPACKE/include/CMakeLists.txt b/LAPACKE/include/CMakeLists.txt
index 034491ea0..9241c5ae3 100644
--- a/LAPACKE/include/CMakeLists.txt
+++ b/LAPACKE/include/CMakeLists.txt
@@ -14,7 +14,7 @@ if(CMAKE_Fortran_COMPILER AND CMAKE_C_COMPILER)
   enable_language(C)
   include(FortranCInterface)
   ## Ensure that the fortran compiler and c compiler specified are compatible
-  FortranCInterface_VERIFY()
+  # FortranCInterface_VERIFY()
   FortranCInterface_HEADER(${LAPACK_BINARY_DIR}/include/lapacke_mangling.h
     MACRO_NAMESPACE "LAPACK_"
     SYMBOL_NAMESPACE "LAPACK_")
@@ -23,4 +23,4 @@ if(NOT FortranCInterface_GLOBAL_FOUND OR NOT FortranCInterface_MODULE_FOUND)
   message(WARNING "Reverting to pre-defined include/lapacke_mangling.h")
   configure_file(lapacke_mangling_with_flags.h.in
                  ${LAPACK_BINARY_DIR}/include/lapacke_mangling.h)
-endif()
\ No newline at end of file
+endif()
```

Then create a `build` directory

```
mkdir build
cd build
cmake .. -DCMAKE_Fortran_COMPILER=lfortran -DCBLAS=no -DLAPACKE=no -DBUILD_TESTING=no -DBUILD_DOUBLE=no -DBUILD_COMPLEX=no -DBUILD_COMPLEX16=no -DLAPACKE_WITH_TMG=no -DCMAKE_Fortran_PREPROCESS=yes -DCMAKE_Fortran_FLAGS="--fixed-form-infer --implicit-interface --generate-object-code --legacy-array-sections" -DCMAKE_INSTALL_LIBDIR=$HOME/ext/lapack
cmake --build . -j --target install
```


----

> Status

<summary>

<details>


```
[....]
[ 54%] Building Fortran object SRC/CMakeFiles/lapack_obj.dir/ssptri.f.o
[ 54%] Building Fortran object SRC/CMakeFiles/lapack_obj.dir/sstev.f.o
[ 54%] Building Fortran object SRC/CMakeFiles/lapack_obj.dir/sptsvx.f.o
[ 54%] Building Fortran object SRC/CMakeFiles/lapack_obj.dir/sstegr.f.o
[ 55%] Building Fortran object SRC/CMakeFiles/lapack_obj.dir/ssbgvd.f.o
[ 55%] Building Fortran object SRC/CMakeFiles/lapack_obj.dir/sspcon.f.o
[ 55%] Building Fortran object SRC/CMakeFiles/lapack_obj.dir/ssptrf.f.o
[ 55%] Building Fortran object SRC/CMakeFiles/lapack_obj.dir/ssycon.f.o
/bin/sh: line 1: 39363 Bus error: 10           /Users/pranavchiku/repos/lfortran/src/bin/lfortran --fixed-form-infer --implicit-interface --generate-object-code --legacy-array-sections -fPIC -c /Users/pranavchiku/repos/lapack/SRC/sgttrs.f -o CMakeFiles/lapack_obj.dir/sgttrs.f.o
[ 55%] Building Fortran object SRC/CMakeFiles/lapack_obj.dir/sspevd.f.o
[ 55%] Building Fortran object SRC/CMakeFiles/lapack_obj.dir/sspev.f.o
[ 55%] Building Fortran object SRC/CMakeFiles/lapack_obj.dir/sspgst.f.o
[ 56%] Building Fortran object SRC/CMakeFiles/lapack_obj.dir/ssprfs.f.o
make[2]: *** [SRC/CMakeFiles/lapack_obj.dir/sgttrs.f.o] Error 138
make[2]: *** Waiting for unfinished jobs....
/bin/sh: line 1: 39402 Bus error: 10           /Users/pranavchiku/repos/lfortran/src/bin/lfortran --fixed-form-infer --implicit-interface --generate-object-code --legacy-array-sections -fPIC -c /Users/pranavchiku/repos/lapack/SRC/shseqr.f -o CMakeFiles/lapack_obj.dir/shseqr.f.o
make[2]: *** [SRC/CMakeFiles/lapack_obj.dir/shseqr.f.o] Error 138
LCOMPILERS_ASSERT failed: /Users/pranavchiku/repos/lfortran/src/libasr/../libasr/asr_utils.h
function Call_t_body(), line number 5942 at 
ASRUtils::check_equal_type(arg_type, orig_arg_type)
ERROR MESSAGE:
ASRUtils::check_equal_type(i32[:], r32[:, :])
[ 57%] Building Fortran object SRC/CMakeFiles/lapack_obj.dir/ssptrs.f.o
[ 57%] Building Fortran object SRC/CMakeFiles/lapack_obj.dir/sstevr.f.o
[ 57%] Building Fortran object SRC/CMakeFiles/lapack_obj.dir/sstevd.f.o
/bin/sh: line 1: 39356 Bus error: 10           /Users/pranavchiku/repos/lfortran/src/bin/lfortran --fixed-form-infer --implicit-interface --generate-object-code --legacy-array-sections -fPIC -c /Users/pranavchiku/repos/lapack/SRC/sggsvp3.f -o CMakeFiles/lapack_obj.dir/sggsvp3.f.o
make[2]: *** [SRC/CMakeFiles/lapack_obj.dir/sggsvp3.f.o] Error 138
/bin/sh: line 1: 39251 Bus error: 10           /Users/pranavchiku/repos/lfortran/src/bin/lfortran --fixed-form-infer --implicit-interface --generate-object-code --legacy-array-sections -fPIC -c /Users/pranavchiku/repos/lapack/SRC/sgees.f -o CMakeFiles/lapack_obj.dir/sgees.f.o
make[2]: *** [SRC/CMakeFiles/lapack_obj.dir/sgees.f.o] Error 138
/bin/sh: line 1: 39224 Bus error: 10           /Users/pranavchiku/repos/lfortran/src/bin/lfortran --fixed-form-infer --implicit-interface --generate-object-code --legacy-array-sections -fPIC -c /Users/pranavchiku/repos/lapack/SRC/sgeesx.f -o CMakeFiles/lapack_obj.dir/sgeesx.f.o
make[2]: *** [SRC/CMakeFiles/lapack_obj.dir/sgeesx.f.o] Error 138
/bin/sh: line 1: 39334 Bus error: 10           /Users/pranavchiku/repos/lfortran/src/bin/lfortran --fixed-form-infer --implicit-interface --generate-object-code --legacy-array-sections -fPIC -c /Users/pranavchiku/repos/lapack/SRC/sggevx.f -o CMakeFiles/lapack_obj.dir/sggevx.f.o
make[2]: *** [SRC/CMakeFiles/lapack_obj.dir/sggevx.f.o] Error 138
/bin/sh: line 1: 39330 Bus error: 10           /Users/pranavchiku/repos/lfortran/src/bin/lfortran --fixed-form-infer --implicit-interface --generate-object-code --legacy-array-sections -fPIC -c /Users/pranavchiku/repos/lapack/SRC/sggev3.f -o CMakeFiles/lapack_obj.dir/sggev3.f.o
make[2]: *** [SRC/CMakeFiles/lapack_obj.dir/sggev3.f.o] Error 138
/bin/sh: line 1: 39238 Bus error: 10           /Users/pranavchiku/repos/lfortran/src/bin/lfortran --fixed-form-infer --implicit-interface --generate-object-code --legacy-array-sections -fPIC -c /Users/pranavchiku/repos/lapack/SRC/sgelsd.f -o CMakeFiles/lapack_obj.dir/sgelsd.f.o
/bin/sh: line 1: 39612 Bus error: 10           /Users/pranavchiku/repos/lfortran/src/bin/lfortran --fixed-form-infer --implicit-interface --generate-object-code --legacy-array-sections -fPIC -c /Users/pranavchiku/repos/lapack/SRC/slarzb.f -o CMakeFiles/lapack_obj.dir/slarzb.f.o
make[2]: *** [SRC/CMakeFiles/lapack_obj.dir/sgelsd.f.o] Error 138
make[2]: *** [SRC/CMakeFiles/lapack_obj.dir/slarzb.f.o] Error 138
/bin/sh: line 1: 39190 Bus error: 10           /Users/pranavchiku/repos/lfortran/src/bin/lfortran --fixed-form-infer --implicit-interface --generate-object-code --legacy-array-sections -fPIC -c /Users/pranavchiku/repos/lapack/SRC/sgbtrf.f -o CMakeFiles/lapack_obj.dir/sgbtrf.f.o
make[2]: *** [SRC/CMakeFiles/lapack_obj.dir/sgbtrf.f.o] Error 138
/bin/sh: line 1: 39349 Bus error: 10           /Users/pranavchiku/repos/lfortran/src/bin/lfortran --fixed-form-infer --implicit-interface --generate-object-code --legacy-array-sections -fPIC -c /Users/pranavchiku/repos/lapack/SRC/sggev.f -o CMakeFiles/lapack_obj.dir/sggev.f.o
make[2]: *** [SRC/CMakeFiles/lapack_obj.dir/sggev.f.o] Error 138
/bin/sh: line 1: 39573 Bus error: 10           /Users/pranavchiku/repos/lfortran/src/bin/lfortran --fixed-form-infer --implicit-interface --generate-object-code --legacy-array-sections -fPIC -c /Users/pranavchiku/repos/lapack/SRC/slarfb_gett.f -o CMakeFiles/lapack_obj.dir/slarfb_gett.f.o
make[2]: *** [SRC/CMakeFiles/lapack_obj.dir/slarfb_gett.f.o] Error 138
/bin/sh: line 1: 39648 Bus error: 10           /Users/pranavchiku/repos/lfortran/src/bin/lfortran --fixed-form-infer --implicit-interface --generate-object-code --legacy-array-sections -fPIC -c /Users/pranavchiku/repos/lapack/SRC/sorgql.f -o CMakeFiles/lapack_obj.dir/sorgql.f.o
make[2]: *** [SRC/CMakeFiles/lapack_obj.dir/sorgql.f.o] Error 138
/bin/sh: line 1: 39234 Bus error: 10           /Users/pranavchiku/repos/lfortran/src/bin/lfortran --fixed-form-infer --implicit-interface --generate-object-code --legacy-array-sections -fPIC -c /Users/pranavchiku/repos/lapack/SRC/sgeev.f -o CMakeFiles/lapack_obj.dir/sgeev.f.o
make[2]: *** [SRC/CMakeFiles/lapack_obj.dir/sgeev.f.o] Error 138
/bin/sh: line 1: 39665 Bus error: 10           /Users/pranavchiku/repos/lfortran/src/bin/lfortran --fixed-form-infer --implicit-interface --generate-object-code --legacy-array-sections -fPIC -c /Users/pranavchiku/repos/lapack/SRC/slatrd.f -o CMakeFiles/lapack_obj.dir/slatrd.f.o
make[2]: *** [SRC/CMakeFiles/lapack_obj.dir/slatrd.f.o] Error 138
/bin/sh: line 1: 39515 Bus error: 10           /Users/pranavchiku/repos/lfortran/src/bin/lfortran --fixed-form-infer --implicit-interface --generate-object-code --legacy-array-sections -fPIC -c /Users/pranavchiku/repos/lapack/SRC/slahr2.f -o CMakeFiles/lapack_obj.dir/slahr2.f.o
make[2]: *** [SRC/CMakeFiles/lapack_obj.dir/slahr2.f.o] Error 138
/bin/sh: line 1: 39663 Bus error: 10           /Users/pranavchiku/repos/lfortran/src/bin/lfortran --fixed-form-infer --implicit-interface --generate-object-code --legacy-array-sections -fPIC -c /Users/pranavchiku/repos/lapack/SRC/slauum.f -o CMakeFiles/lapack_obj.dir/slauum.f.o
make[2]: *** [SRC/CMakeFiles/lapack_obj.dir/slauum.f.o] Error 138
/bin/sh: line 1: 39229 Bus error: 10           /Users/pranavchiku/repos/lfortran/src/bin/lfortran --fixed-form-infer --implicit-interface --generate-object-code --legacy-array-sections -fPIC -c /Users/pranavchiku/repos/lapack/SRC/sgeevx.f -o CMakeFiles/lapack_obj.dir/sgeevx.f.o
make[2]: *** [SRC/CMakeFiles/lapack_obj.dir/sgeevx.f.o] Error 138
/bin/sh: line 1: 39586 Bus error: 10           /Users/pranavchiku/repos/lfortran/src/bin/lfortran --fixed-form-infer --implicit-interface --generate-object-code --legacy-array-sections -fPIC -c /Users/pranavchiku/repos/lapack/SRC/slarft.f -o CMakeFiles/lapack_obj.dir/slarft.f.o
make[2]: *** [SRC/CMakeFiles/lapack_obj.dir/slarft.f.o] Error 138
/bin/sh: line 1: 39420 Bus error: 10           /Users/pranavchiku/repos/lfortran/src/bin/lfortran --fixed-form-infer --implicit-interface --generate-object-code --legacy-array-sections -fPIC -c /Users/pranavchiku/repos/lapack/SRC/slaexc.f -o CMakeFiles/lapack_obj.dir/slaexc.f.o
make[2]: *** [SRC/CMakeFiles/lapack_obj.dir/slaexc.f.o] Error 138
/bin/sh: line 1: 39706 Bus error: 10           /Users/pranavchiku/repos/lfortran/src/bin/lfortran --fixed-form-infer --implicit-interface --generate-object-code --legacy-array-sections -fPIC -c /Users/pranavchiku/repos/lapack/SRC/sormbr.f -o CMakeFiles/lapack_obj.dir/sormbr.f.o
make[2]: *** [SRC/CMakeFiles/lapack_obj.dir/sormbr.f.o] Error 138
/bin/sh: line 1: 39405 Bus error: 10           /Users/pranavchiku/repos/lfortran/src/bin/lfortran --fixed-form-infer --implicit-interface --generate-object-code --legacy-array-sections -fPIC -c /Users/pranavchiku/repos/lapack/SRC/slaqz4.f -o CMakeFiles/lapack_obj.dir/slaqz4.f.o
make[2]: *** [SRC/CMakeFiles/lapack_obj.dir/slaqz4.f.o] Error 138
/bin/sh: line 1: 39676 Bus error: 10           /Users/pranavchiku/repos/lfortran/src/bin/lfortran --fixed-form-infer --implicit-interface --generate-object-code --legacy-array-sections -fPIC -c /Users/pranavchiku/repos/lapack/SRC/sorgbr.f -o CMakeFiles/lapack_obj.dir/sorgbr.f.o
make[2]: *** [SRC/CMakeFiles/lapack_obj.dir/sorgbr.f.o] Error 138
/bin/sh: line 1: 39313 Bus error: 10           /Users/pranavchiku/repos/lfortran/src/bin/lfortran --fixed-form-infer --implicit-interface --generate-object-code --legacy-array-sections -fPIC -c /Users/pranavchiku/repos/lapack/SRC/sgesvdx.f -o CMakeFiles/lapack_obj.dir/sgesvdx.f.o
make[2]: *** [SRC/CMakeFiles/lapack_obj.dir/sgesvdx.f.o] Error 138
/bin/sh: line 1: 39821 Bus error: 10           /Users/pranavchiku/repos/lfortran/src/bin/lfortran --fixed-form-infer --implicit-interface --generate-object-code --legacy-array-sections -fPIC -c /Users/pranavchiku/repos/lapack/SRC/spttrs.f -o CMakeFiles/lapack_obj.dir/spttrs.f.o
make[2]: *** [SRC/CMakeFiles/lapack_obj.dir/spttrs.f.o] Error 138
/bin/sh: line 1: 39655 Bus error: 10           /Users/pranavchiku/repos/lfortran/src/bin/lfortran --fixed-form-infer --implicit-interface --generate-object-code --legacy-array-sections -fPIC -c /Users/pranavchiku/repos/lapack/SRC/sorgrq.f -o CMakeFiles/lapack_obj.dir/sorgrq.f.o
make[2]: *** [SRC/CMakeFiles/lapack_obj.dir/sorgrq.f.o] Error 138
/bin/sh: line 1: 39391 Bus error: 10           /Users/pranavchiku/repos/lfortran/src/bin/lfortran --fixed-form-infer --implicit-interface --generate-object-code --legacy-array-sections -fPIC -c /Users/pranavchiku/repos/lapack/SRC/slaqz3.f -o CMakeFiles/lapack_obj.dir/slaqz3.f.o
make[2]: *** [SRC/CMakeFiles/lapack_obj.dir/slaqz3.f.o] Error 138
/bin/sh: line 1: 39249 Bus error: 10           /Users/pranavchiku/repos/lfortran/src/bin/lfortran --fixed-form-infer --implicit-interface --generate-object-code --legacy-array-sections -fPIC -c /Users/pranavchiku/repos/lapack/SRC/sgelss.f -o CMakeFiles/lapack_obj.dir/sgelss.f.o
make[2]: *** [SRC/CMakeFiles/lapack_obj.dir/sgelss.f.o] Error 138
/bin/sh: line 1: 39542 Bus error: 10           /Users/pranavchiku/repos/lfortran/src/bin/lfortran --fixed-form-infer --implicit-interface --generate-object-code --legacy-array-sections -fPIC -c /Users/pranavchiku/repos/lapack/SRC/slaqr2.f -o CMakeFiles/lapack_obj.dir/slaqr2.f.o
make[2]: *** [SRC/CMakeFiles/lapack_obj.dir/slaqr2.f.o] Error 138
/bin/sh: line 1: 39345 Bus error: 10           /Users/pranavchiku/repos/lfortran/src/bin/lfortran --fixed-form-infer --implicit-interface --generate-object-code --legacy-array-sections -fPIC -c /Users/pranavchiku/repos/lapack/SRC/sgghd3.f -o CMakeFiles/lapack_obj.dir/sgghd3.f.o
make[2]: *** [SRC/CMakeFiles/lapack_obj.dir/sgghd3.f.o] Error 138
/bin/sh: line 1: 39445 Bus error: 10           /Users/pranavchiku/repos/lfortran/src/bin/lfortran --fixed-form-infer --implicit-interface --generate-object-code --legacy-array-sections -fPIC -c /Users/pranavchiku/repos/lapack/SRC/slalsd.f -o CMakeFiles/lapack_obj.dir/slalsd.f.o
make[2]: *** [SRC/CMakeFiles/lapack_obj.dir/slalsd.f.o] Error 138
/bin/sh: line 1: 39571 Bus error: 10           /Users/pranavchiku/repos/lfortran/src/bin/lfortran --fixed-form-infer --implicit-interface --generate-object-code --legacy-array-sections -fPIC -c /Users/pranavchiku/repos/lapack/SRC/slarfb.f -o CMakeFiles/lapack_obj.dir/slarfb.f.o
make[2]: *** [SRC/CMakeFiles/lapack_obj.dir/slarfb.f.o] Error 138
/bin/sh: line 1: 39761 Bus error: 10           /Users/pranavchiku/repos/lfortran/src/bin/lfortran --fixed-form-infer --implicit-interface --generate-object-code --legacy-array-sections -fPIC -c /Users/pranavchiku/repos/lapack/SRC/spbtrf.f -o CMakeFiles/lapack_obj.dir/spbtrf.f.o
make[2]: *** [SRC/CMakeFiles/lapack_obj.dir/spbtrf.f.o] Error 138
/bin/sh: line 1: 39514 Bus error: 10           /Users/pranavchiku/repos/lfortran/src/bin/lfortran --fixed-form-infer --implicit-interface --generate-object-code --legacy-array-sections -fPIC -c /Users/pranavchiku/repos/lapack/SRC/slaqr0.f -o CMakeFiles/lapack_obj.dir/slaqr0.f.o
make[2]: *** [SRC/CMakeFiles/lapack_obj.dir/slaqr0.f.o] Error 138
/bin/sh: line 1: 39521 Bus error: 10           /Users/pranavchiku/repos/lfortran/src/bin/lfortran --fixed-form-infer --implicit-interface --generate-object-code --legacy-array-sections -fPIC -c /Users/pranavchiku/repos/lapack/SRC/slaqr3.f -o CMakeFiles/lapack_obj.dir/slaqr3.f.o
make[2]: *** [SRC/CMakeFiles/lapack_obj.dir/slaqr3.f.o] Error 138
/bin/sh: line 1: 39559 Bus error: 10           /Users/pranavchiku/repos/lfortran/src/bin/lfortran --fixed-form-infer --implicit-interface --generate-object-code --legacy-array-sections -fPIC -c /Users/pranavchiku/repos/lapack/SRC/slaqr4.f -o CMakeFiles/lapack_obj.dir/slaqr4.f.o
make[2]: *** [SRC/CMakeFiles/lapack_obj.dir/slaqr4.f.o] Error 138
/bin/sh: line 1: 39696 Bus error: 10           /Users/pranavchiku/repos/lfortran/src/bin/lfortran --fixed-form-infer --implicit-interface --generate-object-code --legacy-array-sections -fPIC -c /Users/pranavchiku/repos/lapack/SRC/sorm22.f -o CMakeFiles/lapack_obj.dir/sorm22.f.o
make[2]: *** [SRC/CMakeFiles/lapack_obj.dir/sorm22.f.o] Error 138
/bin/sh: line 1: 39323 Bus error: 10           /Users/pranavchiku/repos/lfortran/src/bin/lfortran --fixed-form-infer --implicit-interface --generate-object-code --legacy-array-sections -fPIC -c /Users/pranavchiku/repos/lapack/SRC/sgges3.f -o CMakeFiles/lapack_obj.dir/sgges3.f.o
make[2]: *** [SRC/CMakeFiles/lapack_obj.dir/sgges3.f.o] Error 138
/bin/sh: line 1: 39679 Bus error: 10           /Users/pranavchiku/repos/lfortran/src/bin/lfortran --fixed-form-infer --implicit-interface --generate-object-code --legacy-array-sections -fPIC -c /Users/pranavchiku/repos/lapack/SRC/slatrs3.f -o CMakeFiles/lapack_obj.dir/slatrs3.f.o
make[2]: *** [SRC/CMakeFiles/lapack_obj.dir/slatrs3.f.o] Error 138
/bin/sh: line 1: 39912 Bus error: 10           /Users/pranavchiku/repos/lfortran/src/bin/lfortran --fixed-form-infer --implicit-interface --generate-object-code --legacy-array-sections -fPIC -c /Users/pranavchiku/repos/lapack/SRC/ssptrs.f -o CMakeFiles/lapack_obj.dir/ssptrs.f.o
make[2]: *** [SRC/CMakeFiles/lapack_obj.dir/ssptrs.f.o] Error 138
/bin/sh: line 1: 39288 Bus error: 10           /Users/pranavchiku/repos/lfortran/src/bin/lfortran --fixed-form-infer --implicit-interface --generate-object-code --legacy-array-sections -fPIC -c /Users/pranavchiku/repos/lapack/SRC/sgesdd.f -o CMakeFiles/lapack_obj.dir/sgesdd.f.o
make[2]: *** [SRC/CMakeFiles/lapack_obj.dir/sgesdd.f.o] Error 138
Traceback (most recent call last):
  File "/Users/pranavchiku/repos/lfortran/src/bin/lfortran.cpp", line 2687
    LCompilers::print_stack_on_segfault();
  File "/Users/pranavchiku/repos/lfortran/src/bin/lfortran.cpp", line 2573
    return compile_src_to_object_file(opts.arg_file, outfile, compiler_options.time_report, false,
  File "/Users/pranavchiku/repos/lfortran/src/bin/lfortran.cpp", line 1165
    res = fe.get_llvm3(*asr, lpm, diagnostics, infile);
  File "/Users/pranavchiku/repos/lfortran/src/lfortran/fortran_evaluator.cpp", line 382
    compiler_options, run_fn, infile);
  File "/Users/pranavchiku/repos/lfortran/src/libasr/codegen/asr_to_llvm.cpp", line 10844
    pass_manager.apply_passes(al, &asr, co.po, diagnostics);
  File "/Users/pranavchiku/repos/lfortran/src/libasr/pass/pass_manager.h", line 310
    apply_passes(al, asr, _passes, pass_options, diagnostics);
  File "/Users/pranavchiku/repos/lfortran/src/libasr/pass/pass_manager.h", line 190
    _passes_db[passes[i]](al, *asr, pass_options);
  File "/Users/pranavchiku/repos/lfortran/src/libasr/pass/nested_vars.cpp", line 824
    w.visit_TranslationUnit(unit);
  File "/Users/pranavchiku/repos/lfortran/src/libasr/pass/nested_vars.cpp", line 409
    for (auto &a : x.m_symtab->get_scope()) {
  File "src/libasr/../libasr/asr_base_visitor.h", line 321
  File "src/libasr/../libasr/asr_base_visitor.h", line 31
  File "/Users/pranavchiku/repos/lfortran/src/libasr/pass/nested_vars.cpp", line 488
    transform_stmts(xx.m_body, xx.n_body);
  File "src/libasr/../libasr/asr_expr_call_replacer_visitor.h", line 33
  File "src/libasr/../libasr/asr_base_visitor.h", line 338
  File "src/libasr/../libasr/asr_base_visitor.h", line 67
  File "src/libasr/../libasr/asr_expr_call_replacer_visitor.h", line 385
  File "src/libasr/../libasr/asr_expr_call_replacer_visitor.h", line 33
  File "src/libasr/../libasr/asr_base_visitor.h", line 338
  File "src/libasr/../libasr/asr_base_visitor.h", line 61
  File "src/libasr/../libasr/asr_expr_call_replacer_visitor.h", line 350
  File "src/libasr/../libasr/asr_expr_call_replacer_visitor.h", line 33
  File "src/libasr/../libasr/asr_base_visitor.h", line 338
  File "src/libasr/../libasr/asr_base_visitor.h", line 81
  File "/Users/pranavchiku/repos/lfortran/src/libasr/pass/nested_vars.cpp", line 568
    ASRUtils::Call_t_body(al, xx.m_name, xx.m_args, xx.n_args, x.m_dt,
  File "src/libasr/../libasr/asr_utils.h", line 5988
  Binary file "/usr/lib/system/libsystem_c.dylib", local address: 0x180353a2f
  Binary file "/usr/lib/system/libsystem_pthread.dylib", local address: 0x180446c1f
  Binary file "/usr/lib/system/libsystem_platform.dylib", local address: 0x180477583
Abort: Signal SIGABRT (abort) received

/bin/sh: line 1: 39443 Abort trap: 6           /Users/pranavchiku/repos/lfortran/src/bin/lfortran --fixed-form-infer --implicit-interface --generate-object-code --legacy-array-sections -fPIC -c /Users/pranavchiku/repos/lapack/SRC/slals0.f -o CMakeFiles/lapack_obj.dir/slals0.f.o
make[2]: *** [SRC/CMakeFiles/lapack_obj.dir/slals0.f.o] Error 134
/bin/sh: line 1: 39291 Bus error: 10           /Users/pranavchiku/repos/lfortran/src/bin/lfortran --fixed-form-infer --implicit-interface --generate-object-code --legacy-array-sections -fPIC -c /Users/pranavchiku/repos/lapack/SRC/sgesvd.f -o CMakeFiles/lapack_obj.dir/sgesvd.f.o
make[2]: *** [SRC/CMakeFiles/lapack_obj.dir/sgesvd.f.o] Error 138
make[1]: *** [SRC/CMakeFiles/lapack_obj.dir/all] Error 2
make: *** [all] Error 2
```

</details>

</summary>

--- a/ci/test_third_party_codes.sh
+++ b/ci/test_third_party_codes.sh
@@ -970,133 +970,36 @@ time_section "🧪 Testing Reference-LAPACK v3.12.0 with BUILD_TESTING" '
     # Build BLAS, LAPACK, and test executables
     cmake --build build -j8
 
-    # Helper function to run a LAPACK test and verify results
-    run_lapack_test() {
-        local test_exe=$1
-        local test_input=$2
-        local test_name=$3
-
-        print_subsection "Running LAPACK ${test_name} (${test_exe})"
-        set +e
-        timeout 120 ./bin/${test_exe} < ../TESTING/${test_input} 2>&1 | tee ${test_exe}_${test_input%.in}.out
-        local test_status=${PIPESTATUS[0]}
-        set -e
-
-        echo "${test_exe} ${test_input} exit code: ${test_status}"
-        if [ "${test_status}" -ne 0 ]; then
-            echo "ERROR: ${test_exe} ${test_input} exited with non-zero status"
-            return 1
-        fi
-
-        if grep -qE "failed to pass the threshold" ${test_exe}_${test_input%.in}.out; then
-            echo "ERROR: ${test_exe} ${test_input} threshold failures detected"
-            return 1
-        fi
-
-        local error_messages=$(awk "/error messages recorded/{sum+=\\$1} END{print sum+0}" ${test_exe}_${test_input%.in}.out)
-        echo "${test_exe} ${test_input} error messages recorded: ${error_messages}"
-        if [ "${error_messages}" -ne 0 ]; then
-            echo "ERROR: ${test_exe} ${test_input} recorded error messages"
-            grep -E "error messages recorded" ${test_exe}_${test_input%.in}.out || true
-            return 1
-        fi
-        print_success "${test_name} passed"
-        return 0
-    }
-
-    # Helper for expected failures (XFAIL)
-    run_lapack_xfail() {
-        local test_exe=$1
-        local test_input=$2
-        local test_name=$3
-
-        print_subsection "Running LAPACK ${test_name} (${test_exe}) [XFAIL]"
-        set +e
-        timeout 120 ./bin/${test_exe} < ../TESTING/${test_input} > ${test_exe}_${test_input%.in}.out 2>&1
-        local test_status=$?
-        set -e
-
-        if [ "${test_status}" -eq 0 ]; then
-            echo -e "${GREEN}XPASS: ${test_name} unexpectedly passed!${NC}"
-        else
-            echo -e "${YELLOW}XFAIL: ${test_name} failed as expected (exit ${test_status})${NC}"
-        fi
-    }
-
     cd build
 
-    # Track test results
-    PASSED=0
-    FAILED=0
-    XFAILED=0
-    SKIPPED=0
+    # Run xlintsts (single real linear equations) - the key test
+    print_subsection "Running xlintsts stest.in"
+    set +e
+    timeout 120 ./bin/xlintsts < ../TESTING/stest.in 2>&1 | tee xlintsts_stest.out
+    exit_code=$?
+    set -e
 
-    # Linear equation tests - all 4 real precision variants (all should pass)
-    for test_pair in "xlintsts:stest.in:Single Real Linear Equations" \
-                     "xlintstd:dtest.in:Double Real Linear Equations" \
-                     "xlintstrfs:stest_rfp.in:Single Real RFP Linear Equations" \
-                     "xlintstrfd:dtest_rfp.in:Double Real RFP Linear Equations"; do
-        IFS=: read -r exe input name <<< "$test_pair"
-        if run_lapack_test "$exe" "$input" "$name"; then
-            ((PASSED++))
-        else
-            ((FAILED++))
-            echo "FATAL: Linear equation test failed"
-            exit 1
-        fi
-    done
+    echo "xlintsts exit code: $exit_code"
 
-    # Eigenvalue tests - passing tests
-    for test_pair in "xeigtsts:sep.in:Single SEP" \
-                     "xeigtsts:svd.in:Single SVD" \
-                     "xeigtsts:sec.in:Single SEC" \
-                     "xeigtsts:sgg.in:Single SGG" \
-                     "xeigtsts:ssb.in:Single SSB" \
-                     "xeigtsts:ssg.in:Single SSG" \
-                     "xeigtsts:sbal.in:Single SBAL" \
-                     "xeigtsts:sbak.in:Single SBAK" \
-                     "xeigtsts:sgbal.in:Single SGBAL" \
-                     "xeigtsts:sgbak.in:Single SGBAK" \
-                     "xeigtsts:sbb.in:Single SBB" \
-                     "xeigtstd:sep.in:Double SEP" \
-                     "xeigtstd:svd.in:Double SVD" \
-                     "xeigtstd:dec.in:Double DEC" \
-                     "xeigtstd:dgg.in:Double DGG" \
-                     "xeigtstd:dsb.in:Double DSB" \
-                     "xeigtstd:dsg.in:Double DSG" \
-                     "xeigtstd:dbal.in:Double DBAL" \
-                     "xeigtstd:dbak.in:Double DBAK" \
-                     "xeigtstd:dgbal.in:Double DGBAL" \
-                     "xeigtstd:dgbak.in:Double DGBAK" \
-                     "xeigtstd:dbb.in:Double DBB"; do
-        IFS=: read -r exe input name <<< "$test_pair"
-        if run_lapack_test "$exe" "$input" "$name"; then
-            ((PASSED++))
-        else
-            ((FAILED++))
-            echo "ERROR: Eigenvalue test $name failed unexpectedly"
-            exit 1
-        fi
-    done
+    # Check for failures
+    if [ "$exit_code" -ne 0 ]; then
+        echo "ERROR: xlintsts exited with non-zero status"
+        exit 1
+    fi
 
-    # Eigenvalue tests - expected failures (XFAIL) - runtime issues
-    for test_pair in "xeigtsts:nep.in:Single NEP" \
-                     "xeigtsts:sed.in:Single SED" \
-                     "xeigtsts:sgd.in:Single SGD" \
-                     "xeigtstd:nep.in:Double NEP" \
-                     "xeigtstd:ded.in:Double DED" \
-                     "xeigtstd:dgd.in:Double DGD"; do
-        IFS=: read -r exe input name <<< "$test_pair"
-        run_lapack_xfail "$exe" "$input" "$name"
-        ((XFAILED++))
-    done
+    if grep -qE "failed to pass the threshold" xlintsts_stest.out; then
+        echo "ERROR: threshold failures detected"
+        grep "failed to pass the threshold" xlintsts_stest.out
+        exit 1
+    fi
 
-    # Skipped complex tests (BUILD_COMPLEX=OFF - not built yet)
-    echo -e "${YELLOW}SKIP: Complex tests (BUILD_COMPLEX=OFF)${NC}"
-    ((SKIPPED+=5))
+    # Check for error messages (non-zero count before "error messages recorded")
+    if grep -E "[1-9][0-9]* error messages recorded" xlintsts_stest.out; then
+        echo "ERROR: error messages recorded"
+        exit 1
+    fi
 
-    echo ""
-    print_subsection "LAPACK test summary: ${PASSED} passed, ${FAILED} failed, ${XFAILED} xfail, ${SKIPPED} skipped"
+    print_success "xlintsts stest.in passed"
     cd ../..
 '
 

